pkg/volume/projected/projected.go

```diff
@@ -362,13 +362,9 @@ func (c *projectedVolumeUnmounter) TearDownAt(dir string) error {
 }
 
 func getVolumeSource(spec *volume.Spec) (*v1.ProjectedVolumeSource, bool, error) {
-	var readOnly bool
-	var volumeSource *v1.ProjectedVolumeSource
-
 	if spec.Volume != nil && spec.Volume.Projected != nil {
-		volumeSource = spec.Volume.Projected
-		readOnly = spec.ReadOnly
+		return spec.Volume.Projected, spec.ReadOnly, nil
 	}
 
-	return volumeSource, readOnly, fmt.Errorf("Spec does not reference a projected volume type")
+	return nil, false, fmt.Errorf("Spec does not reference a projected volume type")
```

Comment:
I guess the current behaviour does the same thing? Please let me know if I'm missing something. :)

```
var readOnly bool
var volumeSource *v1.ProjectedVolumeSource
```

will make sure that `nil, false, ...` is returned if not set in the if condition, since the default value (if not initialized) is `false` for the bool and `nil` for the pointer.

Reply:
@nikhita  this function would  always return fmt.Errorf("Spec does not reference a projected volume type")

Reply:
oh of course :woman_facepalming: